react-app/src/components/ProposalDetailScreen/ProposalDetailScreenAPI.ts

```diff
@@ -26,6 +32,52 @@ const calculateTurnout = (tallyResult: Proposal["tallyResult"]) => {
   return turnout.toNumber();
 };
 
+export function useSetProposalReactionMutation(): (
+  proposalId: string,
+  reaction: ReactionType
+) => Promise<ProposalDetailSetReactionMutationMutation> {
+  const mutationFunction = useGraphQLMutation<
+    ProposalDetailSetReactionMutationMutation,
+    ProposalDetailSetReactionMutationMutationVariables
+  >(ProposalDetailSetReactionMutation);
+
+  return useCallback(
+    async (proposalId: string, reaction: ReactionType) => {
+      return mutationFunction({
+        variables: {
+          input: {
+            targetId: `proposal_${proposalId}`,
```

Comment:
client not supposed to format node ids... any easy way to accept node id instead?